script.js

```diff
@@ -1,4 +1,349 @@
+  }
+  //deletes the leaderboardArray,
+  leaderboardArray.length = 0;
+
+  return leaderboardString;
+};
+
+var playKnockoutGame = function () {
+  var outputMsg = `Player ${currentPlayer} has thrown the dice! <br><br>`;
+  var currentPlayerRoll = rollDice();
+  outputMsg += scoreRolls(currentPlayerRoll);
+
+  //if the next player is still available
+  var nextPlayerIndex = currentKnockoutRound.indexOf(currentPlayer) + 1;
+  if (nextPlayerIndex < currentKnockoutRound.length) {
+    currentPlayer = currentKnockoutRound[nextPlayerIndex];
+    outputMsg += `<br><br> It's Player ${currentPlayer}'s turn now!`;
+    return outputMsg;
+  }
+
+  //else it's deliberation mode, which player will be knocked out
+  var losingPlayerScore = Math.min(...resultArray);
+  if (diceMode == LOWEST_MODE) losingPlayerScore = Math.max(...resultArray);
+
+  //this takes the index from the playerarray
+  var losingPlayerIndexFromPlayerArray = resultArray.indexOf(losingPlayerScore);
+
+  //what we need is to take the index from knockoutPlayers array
+  var losingPlayerIndexFromKnockoutArray = knockoutPlayers.indexOf(
+    playerArray[losingPlayerIndexFromPlayerArray]
+  );
+
+  //now we remove the player from the knockoutPlayers array, and update the output msg
+  outputMsg += `<br><br>With a score of ${resultArray[losingPlayerIndexFromPlayerArray]}, Player ${playerArray[losingPlayerIndexFromPlayerArray]} is Eliminated!<br><br>`;
+
+  knockoutPlayers.splice(losingPlayerIndexFromKnockoutArray, 1);
+
+  //meaning there's only 1 player left from the knockoutplayers array, i.e. we've found a winner!
+  if (knockoutPlayers.length == 1) {
+    outputMsg += `Congratulations, Player ${knockoutPlayers[0]}! You win this knockout round!<br><br>`;
+    var winningPlayerIndex = playerArray.indexOf(knockoutPlayers[0]);
+
+    //add leaderboard score
+    highScoreArray[winningPlayerIndex] += 1;
+    outputMsg += leaderboardDisplay();
+
+    //resets the arrays
+    knockoutPlayers = playerArray.slice();
+    outputMsg += `Press roll dice to start a new knockout round!<br><br>`;
+  }
+
+  //resets result array every time someone won
+  resetResultArray();
+  currentKnockoutRound = randomizeCurrentKnockoutRound();
+  currentPlayer = currentKnockoutRound[0];
+  outputMsg += `It's ${currentKnockoutRound[0]} and ${currentKnockoutRound[1]}'s time to battle!<br>`;
+  outputMsg += `It's Player ${currentPlayer}'s turn now!`;
+  return outputMsg;
+};
+
+/***
+ * randomizes the two players selected for knockout game
+ * @returns {Array} 2 players that are in the knockout game, in an array
+ */
+var randomizeCurrentKnockoutRound = function () {
+  if (knockoutPlayers.length > 1) {
+    var randomizedP1Index = Math.floor(Math.random() * knockoutPlayers.length);
+    var randomizedP2Index = Math.floor(Math.random() * knockoutPlayers.length);
+    while (randomizedP1Index == randomizedP2Index) {
+      randomizedP2Index = Math.floor(Math.random() * knockoutPlayers.length);
+    }
```

Comment:
i did this the same way with just a slight tweak, this is a super rare but shocking use case of a do-while loop

```
var randomizedP1Index = Math.floor(Math.random() * knockoutPlayers.length);
var randomizedP2Index;
do {
   randomizedP2Index = Math.floor(Math.random() * knockoutPlayers.length);
 } while (randomizedP1Index == randomizedP2Index)
```